identity-iota/src/did/doc/iota_verification_method.rs

```diff
@@ -156,15 +142,8 @@ impl IotaVerificationMethod {
     IotaDID::check_validity(method.controller())?;
 
     // Ensure the authentication method has an identifying fragment
-    // TODO: validate fragment properly
     if method.id().fragment().is_none() || method.id().fragment().unwrap_or_default().is_empty() {
-      return Err(Error::InvalidDocumentAuthFragment);
-    }
-
-    // Ensure the id and controller are the same - we don't support DIDs
-    // controlled by 3rd parties - yet.
-    if method.id().did().authority() != method.controller().authority() {
-      return Err(Error::InvalidDocumentAuthAuthority);
+      return Err(Error::InvalidMethodMissingFragment);
     }
```

Comment:
This check was unnecessary and obstructive: setting the controller of a verification method does not impact resolution/dereferencing, it is essentially metadata and required for certain controller workarounds.